pinot-query-runtime/src/main/java/org/apache/pinot/query/mailbox/SendingMailbox.java

```diff
@@ -46,4 +49,8 @@ void send(T data)
    * Complete delivery of the current mailbox.
    */
   void complete();
+
+  void waitForFinish(long timeout, TimeUnit unit) throws InterruptedException;
```

Comment:
i dont see `waitForFinish` being called in any place. is it correct?

Reply:
Yeah. will hook the cancel and waitForFinish in followup PRs